blueocean-pipeline-api-impl/src/test/java/io/jenkins/blueocean/rest/impl/pipeline/AbstractRunImplTest.java

```diff
@@ -159,16 +159,19 @@ public void testArtifactZipFileLink() throws Exception {
 
     @Test(timeout = 20000)
     @Issue("JENKINS-44736")
-    public void earlyUnstableStatusShouldReportPunStateAsRunningAndResultAsUnknown() throws Exception {
+    public void earlyUnstableStatusShouldReportRunStateAsRunningAndResultAsUnknown() throws Exception {
         WorkflowJob p = createWorkflowJobWithJenkinsfile("earlyUnstableStatusShouldReportPunStateAsRunningAndResultAsUnknown.jenkinsfile");
 
         Run r = p.scheduleBuild2(0).waitForStart();
 
         String url = "/organizations/jenkins/pipelines/project/runs/" + r.getId() + "/";
         Map m = request().get(url).build(Map.class);
 
-        // Wait 'til we're out of queue and actually on the node.
-        j.waitForMessage("Running on master", r);
+        // Wait until we are up and running before continuing with the test. 
+        while (!m.get("state").equals("RUNNING")) {
```

Comment:
Perhaps add a timeout? Who knows it can wait for ever due to some server error?  Also breaking while loop if state is FINISHED is good too. its possible in 1 sec sleep state transitioned to FINISHED, in that case it can keep running for ever. `while (!m.get("state").equals("RUNNING") && !m.get("state").equals("FINISHED"))`

Reply:
@vivek there is a timeout on the unit test via annotation - is that not enough to fail it?